### ISSUE TITLE
Error: too many values to unpack

### ISSUE THREAD

**Issue by DrChuTang** [autopkg/recipes#191]
I'm attempting to build an over ride recipe thats parent recipe is AdobeReaderUpdates. 

I've updated trust info but I am getting the following error:

> The following failures occurred:

Error in local.jss.Adobe Reader_DCS: Processor: AdobeReaderUpdatesURLProvider: Error: too many values to unpack

I've never seen this error before. Any help is much appreciated. 

Thank you

**Comment by timsutton** [comment]
Let's simplify this first and take your override out of the equation. What happens when you run `autopkg run AdobeReaderUpdates.download -vv` ?

**Comment by DrChuTang** [comment]
Hey Tim,

I get this: 



> computer:RecipeOverrides admin$ autopkg run AdobeReaderUpdates.download -vv
> Processing AdobeReaderUpdates.download...
> WARNING: AdobeReaderUpdates.download is missing trust info and FAIL_RECIPES_WITHOUT_TRUST_INFO is not set. Proceeding...
> AdobeReaderUpdatesURLProvider
> {'Input': {'major_version': u'11', 'os_version': u'10.9'}}
> AdobeReaderUpdatesURLProvider: Found URL http://ardownload.adobe.com/pub/adobe/reader/mac/11.x/11.0.19/misc/AdbeRdrUpd11019.pkg
> {'Output': {'url': 'http://ardownload.adobe.com/pub/adobe/reader/mac/11.x/11.0.19/misc/AdbeRdrUpd11019.pkg',
>             'version': '11.0.19'}}
> URLDownloader
> {'Input': {'filename': u'AdobeReader.pkg',
>            'url': 'http://ardownload.adobe.com/pub/adobe/reader/mac/11.x/11.0.19/misc/AdbeRdrUpd11019.pkg'}}
> URLDownloader: No value supplied for CURL_PATH, setting default value of: /usr/bin/curl
> URLDownloader: No value supplied for CHECK_FILESIZE_ONLY, setting default value of: False
> URLDownloader: Item at URL is unchanged.
> URLDownloader: Using existing /Users/admin/Library/AutoPkg/Cache/com.github.autopkg.download.AdobeReaderUpdates/downloads/AdobeReader.pkg
> {'Output': {'pathname': u'/Users/admin/Library/AutoPkg/Cache/com.github.autopkg.download.AdobeReaderUpdates/downloads/AdobeReader.pkg'}}
> EndOfCheckPhase
> {'Input': {}}
> {'Output': {}}
> Receipt written to /Users/admin/Library/AutoPkg/Cache/com.github.autopkg.download.AdobeReaderUpdates/receipts/AdobeReaderUpdates-receipt-20170224-104544.plist
> 
> Nothing downloaded, packaged or imported.

**Comment by timsutton** [comment]
So the .download recipe works for you.

This means there is an issue with your `local.jss.Adobe Reader_DCS` override or its parent, which judging by your .jss identifier would be a JSS recipe which is not in this repo. Which recipe exactly are you overriding?

**Comment by DrChuTang** [comment]
My parent recipe is com.github.autopkg.download.AdobeReaderUpdates. I'm attempting to over ride it with my local.jss.Adobe Reader_DCS because this local recipe will create my smart group and policy with Casper - our management solution.

**Comment by timsutton** [comment]
So you aren't overriding any of these recipes, your recipe is a direct child of com.github.autopkg.download.AdobeReaderUpdates?

```
Adobe Reader.jss.recipe     jss-recipes           Adobe Reader/Adobe Reader.jss.recipe
Adobe Reader.jss.recipe     novaksam-recipes      Adobe Reader/Adobe Reader.jss.recipe
AdobeReader.jss.recipe      rtrouton-recipes      JSS/AdobeReader.jss.recipe
```

**Comment by DrChuTang** [comment]
You are correct. My recipe is a direct child of com.github.autopkg.download.AdobeReaderUpdates

Bare with me as I've taken over this project. BUT, previously, the ParentRecipe key was set to com.github.jss-recipes.jss.AdobeReader and then there was another key value after that called parent_recipes which was set to com.github.autopkg.download.AdobeReader

I was advised in another thread to use the com.github.autopkg.download.AdobeReaderUpdates so I decided to make a new recipe with that as my parent...

Hope that doesn't confuse you!

**Comment by timsutton** [comment]
The error of "AdobeReaderUpdatesURLProvider: Error: too many values to unpack" suggests that there's something that processor isn't expecting about how it's configured, so it's quite possibly due to your recipe's Inputs. Can you post the full contents of your recipe?

**Comment by DrChuTang** [comment]
Sure, here is my recipe:

<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Identifier</key>
	<string>local.jss.Adobe Reader_DCS</string>
	<key>Input</key>
	<dict>
		<key>CATEGORY</key>
		<string>Digital Media</string>
		<key>GROUP_NAME</key>
		<string>%NAME%-update-smart</string>
		<key>GROUP_TEMPLATE</key>
		<string>SmartGroupTemplate-autoupdateAdobeReader.xml</string>
		<key>LANGUAGE</key>
		<string>English</string>
		<key>MAJOR_VERSION</key>
		<string>11</string>
		<key>NAME</key>
		<string>Adobe Reader</string>
		<key>OS_REQUIREMENTS</key>
		<string>10.11.x,10.10.x,10.9.x</string>
		<key>OS_VERSION</key>
		<string>10.8.0</string>
		<key>POLICY_CATEGORY</key>
		<string>Testing</string>
		<key>POLICY_TEMPLATE</key>
		<string>PolicyTemplate-autoupdateAdobeReader.xml</string>
		<key>SELF_SERVICE_DESCRIPTION</key>
		<string>Views, prints, and comments on PDF documents.</string>
		<key>SELF_SERVICE_ICON</key>
		<string>Adobe Reader.png</string>
	</dict>
	<key>ParentRecipe</key>
	<string>com.github.autopkg.download.AdobeReaderUpdates</string>
	<key>ParentRecipeTrustInfo</key>
	<dict>
		<key>non_core_processors</key>
		<dict>
			<key>AdobeReaderUpdatesURLProvider</key>
			<dict>
				<key>git_hash</key>
				<string>e7a64205c6c027491df847024fe425dcc05b244d</string>
				<key>path</key>
				<string>/Users/admin/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes/AdobeReader/AdobeReaderUpdatesURLProvider.py</string>
				<key>sha256_hash</key>
				<string>bfe64a0b5c0ea0ff50e01b991d712289071b2e06ef9b1c43b8afa9c52ec94a3d</string>
			</dict>
		</dict>
		<key>parent_recipes</key>
		<dict>
			<key>com.github.autopkg.download.AdobeReaderUpdates</key>
			<dict>
				<key>git_hash</key>
				<string>e7a64205c6c027491df847024fe425dcc05b244d</string>
				<key>path</key>
				<string>/Users/admin/Library/AutoPkg/RecipeRepos/com.github.autopkg.recipes/AdobeReader/AdobeReaderUpdates.download.recipe</string>
				<key>sha256_hash</key>
				<string>0d4b6a432294c0cff071c52b9b2545229a45168123ee95bfd65630fed0fe27d8</string>
			</dict>
		</dict>
	</dict>
</dict>
</plist>

**Comment by timsutton** [comment]
Compare your Input keys to that of the parent recipe and eliminate ones that look like they came from the `AdobeReader` recipe (which is very different from the `AdobeReaderUpdates` recipe), and make sure your input's values are consistent with the ones you know are working in `AdobeReaderUpdates.download`.

I'd start with `OS_VERSION` - the parent recipe uses 10.9 and I'm pretty sure the most current updates cut off support for 10.8 (and your use of three components instead of two, used here in the version number, may actually be the real reason why the recipe is giving this error).

**Comment by timsutton** [comment]
Also, it's worth noting that your recipe, while you've named it a "jss" recipe, doesn't actually have any steps in addition to the parent recipe, so unless you're doing something extra here, I don't see how this recipe would do anything in addition to the parent recipe..
